### PR TITLE
Grade api entity

### PR DESCRIPTION
**Pull request by FikretMertGultekin**

### CODE REVIEW

File: homeassistant/components/instructure/canvas_api.py [hunk 1]
```diff
@@ -88,46 +88,97 @@ async def async_get_assignments(self, course_ids: list[str]) -> dict[str, Any]:
...
+        submissions = {}
+        # Grades entity is right now meaningless because we dont get assignment description neither course name etc.
+        # Check GRADES_KEY: CanvasSensorEntityDescription(.... (name_fn and value_fn))
+        for course_id in course_ids:
+            response = await self.async_make_get_request(
+                f"/courses/{course_id}/students/submissions",
+                {"per_page": "50"},
+            )
+            course_submissions = json.loads(response.content.decode("utf-8"))
+            course_submissions = json.loads(response.content.decode("utf-8"))
```
Comment (Caodongying) [inline homeassistant/components/instructure/canvas_api.py]: duplicated

File: homeassistant/components/instructure/const.py [hunk 2]
```diff
@@ -6,3 +6,8 @@
 ANNOUNCEMENTS_KEY: str = "announcements"
 ASSIGNMENTS_KEY: str = "assignments"
 CONVERSATIONS_KEY: str = "conversations"
+GRADES_KEY: str = "grades"
+ANNOUNCEMENT_ENTITY_CONSTANT = 1
```
Comment (Caodongying) [inline homeassistant/components/instructure/const.py:10]: what is this?

**Review by Caodongying — CHANGES_REQUESTED** [review]
Not done yet

File: homeassistant/components/instructure/const.py (hunk 2 above)
Comment (FikretMertGultekin) [reply homeassistant/components/instructure/const.py:10]: this is the constant of entitiy id for announcement. We will have only one announcement entity if it is empty for instance.

File: homeassistant/components/instructure/canvas_api.py (hunk 1 above)
Comment (FikretMertGultekin) [reply homeassistant/components/instructure/canvas_api.py]: will solve it

**Pull request closed without merging**